### ISSUE TITLE
eCSStractor-for-VSCode на линукс

### ISSUE THREAD

**Issue by schklovchikdenis** [yurasovm/eCSStractor-for-VSCode#4]
Добрый день . Поставил ваше расширение на vs code  на линукс. При переносе классов с html в scss, пишет что классы скопированы в clipboard. Но на самом деле в буфере обмена ничего нет. Если в настройках ставить, чтобы классы при переносе открывались в новой вкладке, вкладка с классами открывается. В чем может быть проблема? Спасибо

**Comment by yurasovm** [comment]
Добрый день. В плагине используется api редактора для копирования в буфер обмена. Сам на линуксе не пробовал, но видимо какой-то баг в самом редакторе. Может в обновлениях поправят.

**Comment by IvanNazarov** [comment]
В линукс 2 буфера обмена clipboard и primary. Возможно Вы пытаетесь вставить не из того, в который скопировали?

**Comment by schklovchikdenis** [comment]
Я читал, что в линукса два буфера. При использовании eCSStractor в настройках плагина стоит галочка на clipboard. И при использовании плагина, всплывает окно, что скопировано в clipboard. Но при вставке в Фаил scss, ничего не происходит. Плагин отлично работает и настраивается на винде и маке, а вот на линюксе никак. Были вопросы и у других  пользователей линукса.

**Comment by zdr3zdr** [comment]
В плагине используется
`const ncp = require("copy-paste");`

> node-copy-paste
> 
> A command line utility that allows read/write (i.e copy/paste) access to the system clipboard.It does this by wrapping pbcopy/pbpaste (for OSX), xclip (for Linux and OpenBSD), and clip (for Windows). Currently works with node.js v0.8+.
> 
> xclip – Linux / UNIX Command line clipboard grabber
> 
> You can install xclip using any one of the following method:
> 
> Install xclip under Debian / Ubuntu Linux
> Type the following command at shell prompt:
`$ sudo apt-get install xclip`
> 
> Install xclip under Red hat / CentOS / RHEL / Fedora Linux
> Type the following command at shell prompt (make sure 3rd party repos are activated):
`# yum install xclip`

У меня установка xclip решила проблему.
Ubuntu 18.04.4

**Comment by UrKonAn79** [comment]
> У меня установка xclip решила проблему.

На kubuntu 20.04 тоже работает

**Comment by JasON-official** [comment]
> В плагине используется
> `const ncp = require("copy-paste");`
> 
> > node-copy-paste
> > A command line utility that allows read/write (i.e copy/paste) access to the system clipboard.It does this by wrapping pbcopy/pbpaste (for OSX), xclip (for Linux and OpenBSD), and clip (for Windows). Currently works with node.js v0.8+.
> > xclip – Linux / UNIX Command line clipboard grabber
> > You can install xclip using any one of the following method:
> > Install xclip under Debian / Ubuntu Linux
> > Type the following command at shell prompt:
> > `$ sudo apt-get install xclip`
> > Install xclip under Red hat / CentOS / RHEL / Fedora Linux
> > Type the following command at shell prompt (make sure 3rd party repos are activated):
> > `# yum install xclip`
> 
> У меня установка xclip решила проблему.
> Ubuntu 18.04.4

Благодарю! Помогло Manjaro XFCE

**Comment by N1kk0n** [comment]
> В плагине используется
> `const ncp = require("copy-paste");`
> 
> > node-copy-paste
> > A command line utility that allows read/write (i.e copy/paste) access to the system clipboard.It does this by wrapping pbcopy/pbpaste (for OSX), xclip (for Linux and OpenBSD), and clip (for Windows). Currently works with node.js v0.8+.
> > xclip – Linux / UNIX Command line clipboard grabber
> > You can install xclip using any one of the following method:
> > Install xclip under Debian / Ubuntu Linux
> > Type the following command at shell prompt:
> > `$ sudo apt-get install xclip`
> > Install xclip under Red hat / CentOS / RHEL / Fedora Linux
> > Type the following command at shell prompt (make sure 3rd party repos are activated):
> > `# yum install xclip`
> 
> У меня установка xclip решила проблему.
> Ubuntu 18.04.4

Kubuntu 18.04.4 плагин заработал, но отвалилась комбинация CTRL-V

**Comment by Chippsi** [comment]
Тоже самое было. Pop!_OS(Ubuntu / Debian)
Все решилось двумя командами:
1. sudo apt update
2. sudo apt install xclip

**Comment by Daimzz** [comment]
Добрый день, столкнулся с абсолютно такой же проблемой, но на Windows.

"При переносе классов с html в scss, пишет что классы скопированы в clipboard. Но на самом деле в буфере обмена ничего нет"(c)

Нормально открывает в новой вкладке, но в буфере обмена данных нет...

**Comment by promotezor** [comment]
> Добрый день, столкнулся с абсолютно такой же проблемой, но на Windows.
> 
> "При переносе классов с html в scss, пишет что классы скопированы в clipboard. Но на самом деле в буфере обмена ничего нет"(c)
> 
> Нормально открывает в новой вкладке, но в буфере обмена данных нет...

Поддерживаю, та же самая проблема, даже если поставить не копирование из буфера (clipboard), а в новую вкладку.
Нашлось решение?

**Comment by Daimzz** [comment]
нет, проблема осталась нерешенной. на одном компе все работает отлично, а
на другом проблема осталось нерешенной) возможно переустановка винды и
чистка реестра поможет как-то(почти уверен что поможет).

**Comment by AlisaWonder** [comment]
> В плагине используется
> `const ncp = require("copy-paste");`
> 
> > node-copy-paste
> > A command line utility that allows read/write (i.e copy/paste) access to the system clipboard.It does this by wrapping pbcopy/pbpaste (for OSX), xclip (for Linux and OpenBSD), and clip (for Windows). Currently works with node.js v0.8+.
> > xclip – Linux / UNIX Command line clipboard grabber
> > You can install xclip using any one of the following method:
> > Install xclip under Debian / Ubuntu Linux
> > Type the following command at shell prompt:
> > `$ sudo apt-get install xclip`
> > Install xclip under Red hat / CentOS / RHEL / Fedora Linux
> > Type the following command at shell prompt (make sure 3rd party repos are activated):
> > `# yum install xclip`
> 
> У меня установка xclip решила проблему.
> Ubuntu 18.04.4

Вот, тоже самое было и  установка xclip помогла. Manjaro gnome

**Comment by lgerega** [comment]
после $ sudo apt-get install xclip копирует и вставляет, но в файл css вставляет как в scss, а если нужно без препроцессора работать?

**Comment by alex-shostka** [comment]
> В плагине используется `const ncp = require("copy-paste");`
> 
> > node-copy-paste
> > A command line utility that allows read/write (i.e copy/paste) access to the system clipboard.It does this by wrapping pbcopy/pbpaste (for OSX), xclip (for Linux and OpenBSD), and clip (for Windows). Currently works with node.js v0.8+.
> > xclip – Linux / UNIX Command line clipboard grabber
> > You can install xclip using any one of the following method:
> > Install xclip under Debian / Ubuntu Linux
> > Type the following command at shell prompt:
> > `$ sudo apt-get install xclip`
> > Install xclip under Red hat / CentOS / RHEL / Fedora Linux
> > Type the following command at shell prompt (make sure 3rd party repos are activated):
> > `# yum install xclip`
> 
> У меня установка xclip решила проблему. Ubuntu 18.04.4

На Ubuntu 18 помогло. Спасибо.

**Comment by alex-shostka** [comment]
> после $ sudo apt-get install xclip копирует и вставляет, но в файл css вставляет как в scss, а если нужно без препроцессора работать?

В настройках поагина можно выставить разные способы. С БЭМ и без. С вложенностью и обычный CSS

**Comment by MonstreX** [comment]
Проблема присутствует на WSL2 c Ubuntu. Ни один из способов приведенных выше не работает, к сожалению, все как у всех - копирует в tab, но не в буфер. Желаю автору удачи в борьбе с этим глюком) И спасибо за плагин!

**Comment by anmeshkov** [comment]
Привет. Проблема в том что используется старая библиотека  "copy-paste": "^1.3.0" в ней зависимость от xclip

Для решения проблемы нужно:
1. установить wl-clipboard

```bash
sudo apt install wl-clipboard
```

2: Заменить код в файле linux.js
~/.vscode/extensions/diz.ecsstractor-port-0.0.3/node_modules/copy-paste/platform/

```bash
exports.copy = { command: "wl-copy", args: [] };
exports.paste = { command: "wl-paste", args: [] };
exports.paste.full_command = [ exports.paste.command ].concat(exports.paste.args).join(" ");
exports.encode = str => Buffer.from(str, "utf8");
exports.decode = chunks => {
    if(!Array.isArray(chunks)) chunks = [chunks];
    return Buffer.concat(chunks).toString("utf8");
};
```

3. Перезапустить VSCode
